lib/beaker/hypervisor/docker.rb

```diff
@@ -212,14 +237,29 @@ def provision
           end
         end
 
+        network_settings = container.json['NetworkSettings']
+        host_config = container.json['HostConfig']
+
         @logger.info("Using docker server at #{ip}")
-        port = container.json["NetworkSettings"]["Ports"]["22/tcp"][0]["HostPort"]
 
-        forward_ssh_agent = @options[:forward_ssh_agent] || false
+        port = '22'
+        if host_config['NetworkMode'] == 'bridge' && network_settings['IPAddress'] && !network_settings['IPAddress'].empty?
+          host['ip'] = network_settings['IPAddress']
+        else
+          port = network_settings['Ports']['22/tcp'][0]['HostPort']
+
+          # Update host metadata
+          host['ip'] = (ip == '0.0.0.0') ? '127.0.0.1' : ip
+
+          if (port.to_i < 1024) && (Process.uid != 0)
+            raise "#{host} was given a port less than 1024 but you are not running as root, please try again"
+          end
+        end
```

Comment:
This seems to be triggering the error below:

```
16:09:58 
16:09:58      NoMethodError:
16:09:58        undefined method `[]' for nil:NilClass
16:09:58      # ./lib/beaker/hypervisor/docker.rb:246:in `block in provision'
16:09:58      # ./lib/beaker/hypervisor/docker.rb:131:in `each'
16:09:58      # ./lib/beaker/hypervisor/docker.rb:131:in `provision'
16:09:58      # ./spec/beaker/hypervisor/docker_spec.rb:251:in `block (4 levels) in <module:Beaker>'
16:09:58 
```

Reply:
Looks like something broke further upstream https://puppetcommunity.slack.com/archives/C0W1Y5VL0/p1611698479043200

Reply:
@genebean Prodded you in Slack. Your Jenkins setup might be broken.